language/move-vm/runtime/src/interpreter.rs

```diff
@@ -288,11 +289,17 @@ impl<L: LogContext> Interpreter<L> {
         let native_function = function.get_native()?;
         let result = native_function.dispatch(&mut native_context, ty_args, arguments)?;
         cost_strategy.deduct_gas(result.cost)?;
-        let values = result
+        let return_values = result
             .result
             .map_err(|code| PartialVMError::new(StatusCode::ABORTED).with_sub_status(code))?;
-        for value in values {
-            self.operand_stack.push(value)?;
+        match return_values {
+            NativeReturnValues::None => {}
+            NativeReturnValues::One(value) => self.operand_stack.push(value)?,
+            NativeReturnValues::Many(values) => {
+                for value in values {
+                    self.operand_stack.push(value)?;
```

Comment:
Here can we first extend the operand stack to have a capacity the length of the values, so that it does not increase incrementally? Also I see we are already mapping the error above (line 292). Can we  exit there and only do an append to the operand stack vector here?

Reply:
Right, we should do an append instead of the for loop. Will update it.